### ISSUE TITLE
[R-package] package documentation using roxygen2 7.0.0+

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#2569]
`roxygen2` 7.0.0 was [recently released](https://twitter.com/hadleywickham/status/1194338595202576385?s=19) to CRAN. To close this issue, install `roxygen2` locally and then change the R package's code so that the following works:

```
export CXX=/usr/local/bin/g++-8
export CC=/usr/local/bin/gcc-8
Rscript build_r.R
cd R-package
Rscript -e "roxygen2::roxygenize(load = 'installed')"
```

**Comment by jameslamb** [comment]
closing in favor of being in #2302 . If you'd like to pick this up, comment on the issue and we will open it back up!

**Comment by jameslamb** [comment]
opening this as I think we need to fix it for #2530
